tracer/src/Datadog.Trace/Iast/IastModule.cs

```diff
@@ -600,12 +604,42 @@ private static IastModuleResponse AddWebVulnerability(string? evidenceValue, Int
 
     public static bool AddRequestVulnerabilitiesAllowed()
     {
+        // Check for global budget only
         var currentSpan = (Tracer.Instance.ActiveScope as Scope)?.Span;
         var traceContext = currentSpan?.Context?.TraceContext;
         var isRequest = traceContext?.RootSpan?.Type == SpanTypes.Web;
         return isRequest && traceContext?.IastRequestContext?.AddVulnerabilitiesAllowed() == true;
     }
 
+    private static VulnerabilityStats GetRouteVulnerabilityStats(Span? span)
+    {
+        if (_vulnerabilityStats.Count >= VulnerabilityStatsRouteLimit)
+        {
+            _vulnerabilityStats.Clear(); // Poor man's LRU cache
+        }
+
+        if (span?.Type == SpanTypes.Web)
+        {
+            var route = span.GetTag(Tags.HttpRoute) ?? string.Empty;
+            var method = span.GetTag(Tags.HttpMethod) ?? string.Empty;
+            var key = $"{route}:{method}";
```

Comment:
I'd suggest to you using the `#` as separator and setting the method (GET,POST) at the beginning. 
I think that a route with path param could contain `:` chars, like `/route/:param` so `#` seems safer, and also seems safer and more natural having the method at the beginning.

```suggestion
            var key = $"{method}#{route}";
```